script/tool/test/common/package_looping_command_test.dart

```diff
@@ -252,7 +300,7 @@ void main() {
           command.checkedPackages,
           unorderedEquals(<String>[
             included.path,
-            included.childDirectory('example').path,
+            getExampleDir(included).path,
```

Comment:
How come `getExampleDir` is a function when getting the platform directories is a method?   That seems a bit odd.

Reply:
This is addressed in the `getExampleDir` comment:

```
/// This is deliberately not a method on [RepositoryPackage] since actual tool
/// code should essentially never need this, and instead be using
/// [RepositoryPackage.getExamples] to avoid assuming there's a single example
/// directory. However, needing to construct paths with the example directory
/// is very common in test code.
```

I considered making it a visable-for-testing method instead, but that seemed even more bizarre since it's not for testing `RepositoryPackage`. Making it a free function that's only defined in the test utilities means it'll never even show up as a possibility when people are writing non-test code.

Reply:
Not a huge deal.  It just looked weird compared to the other methods.  Could you have used [extension methods](https://dart.dev/guides/language/extension-methods) in the test utilities?

Reply:
I also considered that, but it violates Flutter style.